homeassistant/components/foscam/entity.py

```diff
@@ -0,0 +1,36 @@
+"""Component providing basic support for Foscam IP cameras."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+from .coordinator import FoscamCoordinator
+
+
+class FoscamEntity(Entity):
+    """Base entity for Foscam camera."""
+
+    def __init__(
+        self,
+        coordinator: FoscamCoordinator,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize the base Foscam entity."""
+
+        self._username = config_entry.data[CONF_USERNAME]
+        self._password = config_entry.data[CONF_PASSWORD]
```

Comment:
What will you use this for if I may ask?

Reply:
Indeed, that should stay in the camera.py